test/async_invoke_test.cpp

```diff
@@ -29,39 +29,61 @@
 using namespace unifex;
 
 TEST(CoInvoke, NoArgumentsNoCaptures) {
-  task<int> t = co_invoke([]() -> task<int> { co_return 42; });
+  task<int> t = async_invoke([]() -> task<int> { co_return 42; });
   std::optional<int> result = sync_wait(std::move(t));
   ASSERT_TRUE(!!result);
   EXPECT_EQ(*result, 42);
 }
 
 TEST(CoInvoke, NoArgumentsWithByValueCaptures) {
   int i = 42;
-  task<int> t = co_invoke([=]() -> task<int> { co_return i; });
+  task<int> t = async_invoke([=]() -> task<int> { co_return i; });
   std::optional<int> result = sync_wait(std::move(t));
   ASSERT_TRUE(!!result);
   EXPECT_EQ(*result, 42);
 }
 
 TEST(CoInvoke, NoArgumentsWithByRefCaptures) {
   int i = 42;
-  task<int> t = co_invoke([&]() -> task<int> { co_return i; });
+  task<int> t = async_invoke([&]() -> task<int> { co_return i; });
   std::optional<int> result = sync_wait(std::move(t));
   ASSERT_TRUE(!!result);
   EXPECT_EQ(*result, 42);
 }
 
 TEST(CoInvoke, WithArgumentsWithByValueCaptures) {
   int i = 42;
-  task<int> t = co_invoke([=](int j) -> task<int> { co_return i + j; }, 58);
+  task<int> t = async_invoke([=](int j) -> task<int> { co_return i + j; }, 58);
   std::optional<int> result = sync_wait(std::move(t));
   ASSERT_TRUE(!!result);
   EXPECT_EQ(*result, 100);
 }
 
 TEST(CoInvoke, WithArgumentsWithByRefCaptures) {
   int i = 42;
-  task<int> t = co_invoke([&](int j) -> task<int> { co_return i + j; }, 58);
+  task<int> t = async_invoke([&](int j) -> task<int> { co_return i + j; }, 58);
+  std::optional<int> result = sync_wait(std::move(t));
+  ASSERT_TRUE(!!result);
+  EXPECT_EQ(*result, 100);
+}
+
+TEST(CoInvoke, WithLvalueArgumentsWithByValueCaptures) {
+  int i = 42;
+  task<int> t = [=]() {
+    int arg = 58;
+    return async_invoke([=](int j) -> task<int> { co_return i + j; }, arg);
+  }();
+  std::optional<int> result = sync_wait(std::move(t));
+  ASSERT_TRUE(!!result);
+  EXPECT_EQ(*result, 100);
+}
+
+TEST(CoInvoke, WithLvalueArgumentsWithByRefCaptures) {
+  int i = 42;
+  task<int> t = [&]() {
+    int arg = 58;
+    return async_invoke([&](int j) -> task<int> { co_return i + j; }, arg);
```

Comment:
Can you also add a test for passing the lambda itself as an lvalue?
e.g. a case where a lambda declared outside of a loop is used multiple times with different args?